### ISSUE TITLE
Invalid version warning on `post` release

### ISSUE THREAD

**Issue by adamtheturtle** [RobertCraigie/pyright-python#308]
I have the latest version of pyright but I'm still getting the version warning

```console
$ pip freeze | grep "pyright"
pyright==1.1.382.post0
$ pyright --version
WARNING: there is a new pyright version available (v1.1.382 -> v1.1.382.post0).
Please install the new version or set PYRIGHT_PYTHON_FORCE_VERSION to `latest`

pyright 1.1.382
```

**Comment by RobertCraigie** [comment]
Thanks for the report, I can reproduce.

**Comment by RobertCraigie** [comment]
@adamtheturtle will have a fix out shortly.

I'm curious if you have any opinions on the version warning itself? I'm [considering removing it](https://github.com/RobertCraigie/pyright-python/issues/302) as I think it hasn't been super helpful but it's difficult to tell if anyone has been getting value out of it.

**Comment by adamtheturtle** [comment]
I don't get value from the warning. It also seems a bit unusual - I have many linters and similar (`mypy`, `ruff`, `vulture`, `interrogate`, `doc8`...) and only `pyright` has this warning.
